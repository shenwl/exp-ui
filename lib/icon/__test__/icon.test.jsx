import React from 'react';
import * as renderer from 'react-test-renderer';
import Icon from '../index';
import { mount } from 'enzyme';

describe('测试Icon组件', () => {
  it('UI测试', () => {
    const json = renderer.create(<Icon name='wechat' />).toJSON()
    expect(json).toMatchSnapshot()
  })

  it('onclick', () => {
    const fn = jest.fn();
    const comp = mount(<Icon name='wechat' onClick={fn} />);
    comp.find('svg').simulate('click');
    expect(fn).toBeCalled();
  })
})
