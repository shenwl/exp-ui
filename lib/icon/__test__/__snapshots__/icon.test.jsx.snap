// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon render successfully 1`] = `
<svg
  className="exp__icon"
>
  <use
    xlinkHref="#wechat"
  />
</svg>
`;
